src/portal/src/Portal.js

```diff
@@ -1,38 +1,50 @@
-import { Component } from 'react'
+import { useState, useEffect, memo } from 'react'
 import canUseDom from 'dom-helpers/util/inDOM'
 import ReactDOM from 'react-dom'
 import PropTypes from 'prop-types'
 
-let portalContainer
+const initializePortal = () => {
+  if (!canUseDom) {
+    return null
+  }
+
+  const portalContainer = document.createElement('div')
+  portalContainer.setAttribute('evergreen-portal-container', '')
+  return portalContainer
+}
 
-export default class Portal extends Component {
-  constructor() {
-    super()
+const initializeEl = () => {
+  if (!canUseDom) {
+    return null
+  }
+
+  return document.createElement('div')
+}
 
-    // This fixes SSR
-    if (!canUseDom) return
+const Portal = memo(({ children }) => {
+  const [portalContainer] = useState(initializePortal)
+  const [el] = useState(initializeEl)
```

Comment:
Wouldn't this require `initializePortal()` and `initializeEl()` to be called?

Reply:
Also, won't this cause `initializePortal()` to get called every time there's a Portal component mounted? We don't want multiple divs for the portal, just one.

Reply:
For your first comment, yeah I think to have the same exact behavior as the constructor we need to call the functions. I didn't know passing a function would lazily set the state until now!

Reply:
<img width="1226" alt="Screen Shot 2020-05-29 at 08 42 28" src="https://user-images.githubusercontent.com/13311268/83278787-f833f400-a188-11ea-8cd1-9f19aa5d6ec7.png">

that looks better :)